lock-impl/src/test/java/com/palantir/lock/impl/AdjustableBackgroundTaskTest.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.palantir.refreshable.Refreshable;
+import com.palantir.refreshable.SettableRefreshable;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.jmock.lib.concurrent.DeterministicScheduler;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class AdjustableBackgroundTaskTest {
+    private static final Duration DEFAULT_INTERVAL = Duration.ofSeconds(10);
+    private final AtomicInteger field = new AtomicInteger(0);
+    private final SettableRefreshable<Boolean> shouldRun = Refreshable.create(false);
+    private final SettableRefreshable<Duration> interval = Refreshable.create(DEFAULT_INTERVAL);
+    private final DeterministicScheduler scheduledExecutor = new DeterministicScheduler();
+    private final AtomicInteger numCallsToShouldRunSupplier = new AtomicInteger();
```

Comment:
Nit: I'd make `numCallsToShouldRunSupplier` & `numCallsToIntervalSupplier` equivalent, in that, they should either both be `new AtomicInteger()` or `new AtomicInteger(0)`; under the hood it doesn't matter, as the default value is 0 anyways.

Reply:
ah, this was a typo